client/me/developer/main.jsx

```diff
@@ -128,35 +109,57 @@ class Developer extends Component {
 
 	render() {
 		return (
-			<Main className="developer" wideLayout>
-				<PageViewTracker path="/me/developer" title="Me > Developer" />
-				<ReauthRequired twoStepAuthorization={ twoStepAuthorization } />
-				<NavigationHeader
-					navigationItems={ [] }
-					title={ this.props.translate( 'Developer Features' ) }
-					subtitle={ this.props.translate(
-						'Take WordPress.com further with early access to new developer features.'
+			<>
+				<Main className="developer" wideLayout>
+					<PageViewTracker path="/me/developer" title="Me > Developer" />
+					<ReauthRequired twoStepAuthorization={ twoStepAuthorization } />
+					<NavigationHeader
+						navigationItems={ [] }
+						title={ this.props.translate( 'Developer Features' ) }
+						subtitle={ this.props.translate(
+							'Take WordPress.com further with early access to new developer features.'
+						) }
+						className="developer__header"
+					/>
+
+					<form onChange={ this.props.submitForm }>
+						<FormFieldset
+							className={ classnames( 'developer__is_dev_account-fieldset', {
+								'is-loading': this.props.isFetchingUserSettings,
+							} ) }
+						>
+							<ToggleControl
+								disabled={ this.props.isFetchingUserSettings || this.props.isUpdatingUserSettings }
+								checked={ this.props.getSetting( 'is_dev_account' ) }
+								onChange={ this.handleToggleIsDevAccount }
+								label={ getIAmDeveloperCopy( this.props.translate ) }
+							/>
+						</FormFieldset>
+					</form>
+
+					<DeveloperFeatures />
+				</Main>
+
+				{ this.state.isVisibleDevSurveyNotice &&
+					ReactDOM.createPortal(
+						<DevSurveyNotice
+							localeSlug={ this.props.currentUser.localeSlug }
+							onOk={ () => {
+								recordTracksEvent( 'calypso_me_developer_survey_clicked' );
+
+								this.hideDevSurveyNotice();
+							} }
+							onClose={ ( remindTimeInSeconds ) => {
+								recordTracksEvent( 'calypso_me_developer_survey_dismissed' );
```

Comment:
```suggestion
								recordTracksEvent( 'calypso_me_developer_survey_dismissed', { context: <ELEMENT_CLICKED> } );
```

It'd be great if we could add extra event prop differentiate the elements that users clicked on to dismiss the prompt. For example:
* `close-button`
* `remind-later-button`
* `backdrop`

Reply:
I'm not sure we need that level of granularity but I'll let you decide 🙂.

Reply:
>I'm not sure we need that level of granularity but I'll let you decide 

I'm not sure either, but that is why I proposed adding event details, instead of introducing a new event for each type of dismissal. It's nice to have the option to filter down by context **if** needed.